spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/TestSparkV2Filters.java

```diff
@@ -48,145 +48,150 @@ public void testV2Filters() {
-          Expression expectedEqNullSafe2 = Expressions.equal(unquoted, 1);
-          Expression actualEqNullSafe2 = SparkV2Filters.convert(eqNullSafe2);
-          Assert.assertEquals(
-              "EqualNullSafe must match",
-              expectedEqNullSafe2.toString(),
-              actualEqNullSafe2.toString());
-
-          LiteralValue str =
-              new LiteralValue(UTF8String.fromString("iceberg"), DataTypes.StringType);
-          org.apache.spark.sql.connector.expressions.Expression[] attrAndStr =
-              new org.apache.spark.sql.connector.expressions.Expression[] {namedReference, str};
-          Predicate startsWith = new Predicate("STARTS_WITH", attrAndStr);
-          Expression expectedStartsWith = Expressions.startsWith(unquoted, "iceberg");
-          Expression actualStartsWith = SparkV2Filters.convert(startsWith);
-          Assert.assertEquals(
-              "StartsWith must match", expectedStartsWith.toString(), actualStartsWith.toString());
-
-          Predicate in = new Predicate("IN", attrAndValue);
-          Expression expectedIn = Expressions.in(unquoted, 1);
-          Expression actualIn = SparkV2Filters.convert(in);
-          Assert.assertEquals("In must match", expectedIn.toString(), actualIn.toString());
-
-          Predicate and = new And(lt1, eq1);
-          Expression expectedAnd = Expressions.and(expectedLt1, expectedEq1);
-          Expression actualAnd = SparkV2Filters.convert(and);
-          Assert.assertEquals("And must match", expectedAnd.toString(), actualAnd.toString());
-
-          org.apache.spark.sql.connector.expressions.Expression[] attrAndAttr =
-              new org.apache.spark.sql.connector.expressions.Expression[] {
-                namedReference, namedReference
-              };
-          Predicate invalid = new Predicate("<", attrAndAttr);
-          Predicate andWithInvalidLeft = new And(invalid, eq1);
-          Expression convertedAnd = SparkV2Filters.convert(andWithInvalidLeft);
-          Assert.assertEquals("And must match", convertedAnd, null);
-
-          Predicate or = new Or(lt1, eq1);
-          Expression expectedOr = Expressions.or(expectedLt1, expectedEq1);
-          Expression actualOr = SparkV2Filters.convert(or);
-          Assert.assertEquals("Or must match", expectedOr.toString(), actualOr.toString());
-
-          Predicate orWithInvalidLeft = new Or(invalid, eq1);
-          Expression convertedOr = SparkV2Filters.convert(orWithInvalidLeft);
-          Assert.assertEquals("Or must match", convertedOr, null);
-
-          Predicate not = new Not(lt1);
-          Expression expectedNot = Expressions.not(expectedLt1);
-          Expression actualNot = SparkV2Filters.convert(not);
-          Assert.assertEquals("Not must match", expectedNot.toString(), actualNot.toString());
-        });
+    attrMap.forEach(this::testV2Filter);
+  }
+
+  private void testV2Filter(String quoted, String unquoted) {
```

Comment:
I think this is a bit hacky to avoid the length warning, I think @RussellSpitzer may have meant individual test methods, like:

```
@Test
public void testIsNullV2Filter() {
}

@Test
public void testIsNotNullV2Filter {
}
```

Maybe we can revert it to what you had, and we can make a cleanup issue for this?

Reply:
Oh sorry for the misunderstood. Reverted and will address it in a follow-up.